concent_api/core/payments/bankster.py

```diff
@@ -477,18 +477,19 @@ def discard_claim(deposit_claim: DepositClaim) -> bool:
     assert isinstance(deposit_claim, DepositClaim)
 
     with transaction.atomic(using='control'):
-        DepositAccount.objects.select_for_update().get(
-            pk=deposit_claim.payer_deposit_account_id
-        )
+        try:
+            DepositAccount.objects.select_for_update().get(
+                pk=deposit_claim.payer_deposit_account_id
+            )
+        except DepositAccount.DoesNotExist:
```

Comment:
If this account doesn't exist, how did we get so far?

Reply:
It exists, in the first version of my code there was no such check, but as is was added later in wrong place I have moved it into correct one. It can be here or not, it should not make much difference. I would keep it as sanity check and good practice.

Reply:
It can always happen that it was removed in some other way, through admin panel, manually in database etc. So there is a case. 

How should `assert` looks here then? It can't be verified without making a query.

Reply:
> It can always happen that it was removed in some other way, through admin panel, manually in database etc. So there is a case.

We don't support manual database modifications. And if this happens without such modifications, it's a bug. It's fine to crash then. If we wanted to work with an assumption that the database can be modified in an arbitrary way, we'd have to stuff all our code with such defensive checks and I it would make the code impossible to work with.

Our basic assumption is that the database is over Concent's total control.

We might want to handle manual modifications in some cases where they're likely but this is not one of them.

> How should `assert` looks here then? It can't be verified without making a query.

What's wrong about doing a query?

But anyway, I did not mean that an `assert` is needed here. I wanted to say that if you're worried about the object not being there, you should check it with an `assert` because it makes your intentions clear (you assume that this is never supposed to happen) rather than by adding a handler (which implies that it's something that can happen and we need to handle it).

Reply:
You could do
``` python
except DepositAccount.DoesNotExist:
    assert False
```

Reply:
:+1: done